src/ucp/proto/proto_common.c

```diff
@@ -147,17 +155,36 @@ size_t ucp_proto_common_get_iface_attr_field(const uct_iface_attr_t *iface_attr,
     return *(const size_t*)UCS_PTR_BYTE_OFFSET(iface_attr, field_offset);
 }
 
+static int
+ucp_proto_common_adjust_max_frag(const ucp_proto_common_init_params_t *params)
+{
+    return (params->send_op == UCT_EP_OP_AM_SHORT) ||
+        (params->send_op == UCT_EP_OP_AM_BCOPY) ||
```

Comment:
looking at this, maybe better define new flag in `ucp_proto_single_init_params_t.super.flags`, which would indicate whether max_frag_size adjustment is needed? Then we can set this flag in `proto_init` functions and will have to check for just 1 flag